risc0/zkvm/src/host/client/env.rs

```diff
@@ -194,6 +207,21 @@ impl<'a> ExecutorEnvBuilder<'a> {
         self
     }
 
+    /// Set powers of 2 cycles for keccak proofs
+    ///
+    /// Lowering this value will reduce the memory consumption of the prover. Memory consumption is
+    /// roughly linear with the segment size, so lowering this value by 1 will cut memory
+    /// consumpton by about half.
+    ///
+    /// The default value is chosen to be performant on commonly used hardware. Tuning this value,
+    /// either up or down, may result in better proving performance.
+    ///
+    /// Given value must be within [risc0_circuit_keccak::KECCAK_PO2_RANGE].
+    pub fn keccak_po2(&mut self, limit: u32) -> &mut Self {
```

Comment:
This is probably the "max" po2. Seems like we'll need to plumb this thru client/server, but it'd be nice if there was a way to avoid that.

Reply:
executor env has an env_vars field. we could porbably stuff it in there